File: src/main/java/org/zenbot/szolnok/timetable/dao/BusStopRepository.java
package org.zenbot.szolnok.timetable.dao;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.zenbot.szolnok.timetable.domain.BusStopWithBuses;

import java.util.Optional;

public interface BusStopRepository extends MongoRepository<BusStopWithBuses, String> {

    Optional<BusStopWithBuses> findByBusStopName(String s);
}

File: src/main/kotlin/org/zenbot/szolnok/timetable/dao/BusStopRepository.kt
package org.zenbot.szolnok.timetable.dao

import org.springframework.data.mongodb.repository.MongoRepository
import org.zenbot.szolnok.timetable.domain.BusStopWithBuses

import java.util.Optional

interface BusStopRepository : MongoRepository<BusStopWithBuses, String> {

    fun findByBusStopName(s: String): Optional<BusStopWithBuses>
}
